src/FormElement/ScheduleElement.php

```diff
@@ -0,0 +1,628 @@
+        ];
+
+        $this->customFrequencies = array_slice($this->regulars, 2);
+        unset($this->customFrequencies[RRule::QUARTERLY]);
+
+        $this->advanced = [
+            static::CUSTOM_EXPR => $this->translate('Custom…'),
+            static::CRON_EXPR   => $this->translate('Cron Expression…')
+        ];
+    }
+
+    /**
+     * Get whether this element is rendering a cron expression
+     *
+     * @return bool
+     */
+    public function hasCronExpression(): bool
+    {
+        return $this->getFrequency() === static::CRON_EXPR;
+    }
+
+    /**
+     * Get the frequency of this element
+     *
+     * @return string
+     */
+    public function getFrequency(): string
+    {
+        return parent::getValue('frequency', $this->frequency);
+    }
+
+    /**
+     * Set the custom frequency of this schedule element
+     *
+     * @param string $frequency
+     *
+     * @return $this
+     */
+    public function setFrequency(string $frequency): self
+    {
+        if (
+            $frequency !== static::NO_REPEAT
+            && ! isset($this->regulars[$frequency])
+            && ! isset($this->advanced[$frequency])
+        ) {
+            throw new InvalidArgumentException(sprintf('Invalid frequency provided: %s', $frequency));
+        }
+
+        $this->frequency = $frequency;
+
+        return $this;
+    }
+
+    /**
+     * Get custom frequency of this element
+     *
+     * @return ?string
+     */
+    public function getCustomFrequency(): ?string
+    {
+        return parent::getValue('custom-frequency', $this->customFrequency);
+    }
+
+    /**
+     * Set custom frequency of this element
+     *
+     * @param string $frequency
+     *
+     * @return $this
+     */
+    public function setCustomFrequency(string $frequency): self
+    {
+        if (! isset($this->customFrequencies[$frequency])) {
+            throw new InvalidArgumentException(sprintf('Invalid custom frequency provided: %s', $frequency));
+        }
+
+        $this->customFrequency = $frequency;
+
+        return $this;
+    }
+
+    /**
+     * Set start time of the parsed expressions
+     *
+     * @param DateTime $start
+     *
+     * @return $this
+     */
+    public function setStart(DateTime $start): self
+    {
+        $this->start = $start;
+
+        // Forward the start time update to the sub elements as well!
+        $this->weeklyField->setDefault($start->format('D'));
+        $this->annuallyFields->setDefault($start->format('M'));
+        $this->monthlyFields
+            ->setDefault((int) $start->format('j'))
+            ->setAvailableFields((int) $start->format('t'));
+
+        return $this;
+    }
+
+    /**
+     * Set the end time of this schedule element
+     *
+     * @param DateTime $end
+     *
+     * @return $this
+     */
+    public function setEnd(DateTime $end): self
+    {
+        $this->end = $end;
+
+        return $this;
+    }
+
+    public function getValue($name = null, $default = null)
```

Comment:
I suppose, if `$name` is given, the named element's value should be returned. `getElement($name)->getValue()` is also possible, so why shouldn't this.